### DISCUSSION TITLE
Is it legal to use external sources of data?

### DISCUSSION THREAD

**Discussion by szeke** [snap-stanford/ogb#147] [WikiKG90M-LSC]
Can we augment the provided data with additional data from other sources?

**Comment by weihua916** [comment]
We do not allow the use of external data. See the rule (Use of External Data, Public Data Source) [here](https://ogb.stanford.edu/kddcup2021/rules/).

**Reply by szeke** [reply]
Ah, thx

**Reply by zhengsx** [reply]
Hi Weihua. does the ogb team have some technical method to detect the usage of external data, except of the submission of source code. It may not be effective for external data use detection sometime. Does the team consider to reproduce the result from scratch?

**Reply by weihua916** [reply]
We will use the source code, which should reproduce the result from scratch. And yes, either we or the community will try to reproduce the result using the provided code. If any misconduct is found after the winner announcement, we will disqualify the winner (which hopefully won't happen...).

**Comment by schwomas** [comment]
Out of curiosity: Why not supply the raw texts? Is this to align with previous publications?
(Training a text-embedder in the loop could lead to better solutions, i guess).

**Reply by weihua916** [reply]
Good point! The primary reason is to keep the data anonymous. See public data source [here](https://ogb.stanford.edu/kddcup2021/rules/).

As we agree that training a feature extractor end-to-end is beneficial, we are thinking about releasing the raw text at some point in the future (after the contest).

**Reply by cj2001** [reply]
Just throwing in my opinion...

...having the raw text would be of tremendous community benefit.  It would be very useful for the challenge to try and get a very rough sense of the quality of a handful of predicted links just by quickly looking at them.  Further, as a community education tool and development tool it would be incredibly helpful for users to be able to work with a knowledge graph that makes intuitive sense (rather than just a series of unrelated integers).  

I totally get that this is not the intent of the competition and there are different goals at work here.  I would just like to encourage you to someday (hopefully soon!) release that raw text for the benefit of the community.

Thank you!
